pkg/storage/allocator_test.go

```diff
@@ -685,9 +694,164 @@ func TestAllocatorRebalance(t *testing.T) {
 	}
 }
 
+// TestAllocatorRebalanceTarget could help us to verify whether we'll rebalance to a target that
+// we'll immediately remove.
+func TestAllocatorRebalanceTarget(t *testing.T) {
```

Comment:
Can you confirm that this test passes with your change but fails without it?

Reply:
Yes, I could confirm that.